compiler/luci/pass/src/DecomposeSoftmaxPass.test.cpp

```diff
@@ -0,0 +1,230 @@
+  ASSERT_NE(nullptr, name);
+}
+
+/**
+ *  Decomposed graph looks like below.
+ *
+ *      [CircleNode]   [CircleConst(=-1)]
+ *          |    \       /           |
+ *          |     \     /            |
+ *          | [CircleReduceMax]      |
+ *          |    /                   |
+ *          |   /                    |
+ *          |  /                     |
+ *        [Sub]                      |
+ *          |                        |
+ *          | [CircleConst(=beta)]   |
+ *          |   /                    |
+ *          |  /                     |
+ *        [Mul]  (if beta != 1)      |
+ *          |                        |
+ *        [Exp]                      |
+ *          | \                      |
+ *          |  \                     |
+ *          |  [CircleSum]-----------+
+ *          |  /
+ *          | /
+ *        [Div]
+ *          |
+ *          |
+ *      [CircleNode]
+ */
+TEST_F(DecomposeSoftmaxPass, simple_test)
+{
+  auto ret = _pass.run(&_softmax_g._g);
+  EXPECT_TRUE(ret);
+
+  auto div = dynamic_cast<luci::CircleDiv *>(_softmax_g._output->from());
+  EXPECT_NE(nullptr, div);
+
+  auto exp = dynamic_cast<luci::CircleExp *>(div->x());
+  EXPECT_NE(nullptr, exp);
+
+  auto sum = dynamic_cast<luci::CircleSum *>(div->y());
+  EXPECT_NE(nullptr, sum);
+
+  auto exp_1 = dynamic_cast<luci::CircleExp *>(sum->input());
+  EXPECT_EQ(exp, exp_1);
+
+  auto indices = dynamic_cast<luci::CircleConst *>(sum->reduction_indices());
+  EXPECT_NE(nullptr, indices);
+  EXPECT_EQ(indices->dtype(), loco::DataType::S32);
+  EXPECT_EQ(indices->size<loco::DataType::S32>(), 1);
+  EXPECT_EQ(indices->scalar<loco::DataType::S32>(), -1);
+
+  auto mul = dynamic_cast<luci::CircleMul *>(exp->x());
+  EXPECT_NE(nullptr, mul);
+
+  auto sub = dynamic_cast<luci::CircleSub *>(mul->x());
+  EXPECT_NE(nullptr, sub);
+
+  auto beta = dynamic_cast<luci::CircleConst *>(mul->y());
+  EXPECT_NE(nullptr, beta);
+  EXPECT_EQ(beta->dtype(), loco::DataType::FLOAT32);
+  EXPECT_EQ(beta->size<loco::DataType::FLOAT32>(), 1);
+
+  auto input = dynamic_cast<luci::CircleInput *>(sub->x());
+  EXPECT_NE(nullptr, input);
+
+  auto max = dynamic_cast<luci::CircleReduceMax *>(sub->y());
+  EXPECT_NE(nullptr, max);
+
+  auto indices_1 = dynamic_cast<luci::CircleConst *>(max->reduction_indices());
+  EXPECT_NE(nullptr, indices_1);
+  EXPECT_EQ(indices, indices_1);
+
+  auto input_1 = dynamic_cast<luci::CircleInput *>(max->input());
+  EXPECT_NE(nullptr, input_1);
+  EXPECT_EQ(input, input_1);
+}
+
+TEST_F(DecomposeSoftmaxPass, check_last_node)
+{
+  auto ret = _pass.run(&_softmax_g._g);
+  EXPECT_TRUE(ret);
+
+  auto softmax = dynamic_cast<luci::CircleSoftmax *>(_softmax_g._output->from());
+  EXPECT_EQ(nullptr, softmax);
+}
+
+TEST_F(DecomposeSoftmaxPass, wrong_condition_NEG)
+{
+  auto ret = _pass.run(&_softmax_int32_g._g);
+  EXPECT_FALSE(ret);
+
+  auto softmax = dynamic_cast<luci::CircleSoftmax *>(_softmax_g._output->from());
+  EXPECT_NE(nullptr, softmax);
+}
```

Comment:
if possible, please add more NEG tests so that num_of(positive) <= num_of(negative)